cmd/skopeo/seed.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/containers/image/torrent"
+	"github.com/containers/image/transports/alltransports"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+func seedHandler(c *cli.Context) error {
+	policyContext, err := getPolicyContext(c)
+	if err != nil {
+		return fmt.Errorf("Error loading trust policy: %v", err)
+	}
+	defer policyContext.Destroy()
+
+	trackers := c.StringSlice("torrent-trackers")
+
+	ctx, err := contextFromGlobalOptions(c, "")
+	if err != nil {
+		return err
+	}
+	ctx.DockerTorrentTrackers = trackers
+
+	if len(c.Args()) < 2 || len(c.Args())%2 != 0 {
+		cli.ShowCommandHelp(c, "seed")
+		return errors.New("Invalid number of arguments")
+	}
+
+	debug := logrus.GetLevel() == logrus.DebugLevel
+
+	client, err := torrent.MakeClient(ctx, debug, true)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	for off := 0; off < len(c.Args()); off = off + 2 {
+		ref, err := alltransports.ParseImageName(c.Args()[off])
+		if err != nil {
+			return fmt.Errorf("Invalid source image %s: %v", c.Args()[off], err)
+		}
+
+		refSrc, err := alltransports.ParseImageName(c.Args()[off+1])
+		if err != nil {
+			return fmt.Errorf("Invalid storage image %s: %v", c.Args()[off+1], err)
+		}
+
+		if err := client.Seed(context.Background(), ctx, ref, refSrc); err != nil {
+			return err
+		}
+	}
+	if debug {
+		go func() {
+			for {
+				time.Sleep(time.Second * 10)
+				client.WriteStatus(os.Stderr)
+			}
+		}()
+	}
+
+	s := make(chan os.Signal)
+	signal.Notify(s, os.Interrupt, syscall.SIGTERM)
+	<-s
+	logrus.Info("got signal, shut down")
+	client.Close()
+	os.Exit(1)
```

Comment:
I am using this so that we don't leave temporary files around, as `skopeo seed` never exits, but if we can read directly from the `dir:` storage so that we don't need another copy, then it should be fine to not handle the signal here

Reply:
I am perfectly fine with doing all necessary cleanup; all I wanted to suggest is that this should just return `nil`, let all other cleanups, if any, happen, and have the process return the default exit status 0.

Reply:
> I am perfectly fine with doing all necessary cleanup

… and waiting for a signal.